webdriver/tests/classic/delete_session/delete.py

```diff
@@ -9,27 +9,39 @@ def delete_session(session):
 
 
 def test_null_response_value(session):
-    response = delete_session(session)
-    value = assert_success(response)
-    assert value is None
+    try:
+        response = delete_session(session)
+        value = assert_success(response)
+        assert value is None
 
-    # Need an explicit call to session.end() to notify the test harness
-    # that a new session needs to be created for subsequent tests.
-    session.end()
+    finally:
+        response = session.transport.send(
+            "GET", f"session/{session.session_id}/alert/text"
+        )
+        assert_error(response, "invalid session id")
+
+        # Need an explicit call to session.end() to notify the test harness
+        # that a new session needs to be created for subsequent tests.
+        session.end()
 
 
 def test_accepted_beforeunload_prompt(session, url):
     session.url = url("/webdriver/tests/support/html/beforeunload.html")
 
     session.find.css("input", all=False).send_keys("foo")
 
-    response = delete_session(session)
-    assert_success(response)
-
-    # A beforeunload prompt has to be automatically accepted, and the session deleted
-    with pytest.raises(error.InvalidSessionIdException):
-        session.alert.text
-
-    # Need an explicit call to session.end() to notify the test harness
-    # that a new session needs to be created for subsequent tests.
-    session.end()
+    try:
+        response = delete_session(session)
+        assert_success(response)
+
+        # A beforeunload prompt has to be automatically accepted, and the
+        # session deleted.
+        response = session.transport.send(
+            "GET", f"session/{session.session_id}/alert/text"
+        )
+        assert_error(response, "invalid session id")
```

Comment:
Also can I ask why you changed those lines? They are fine and to explain why, we usually do only use the `transport.send()` method for the command under test but not for other commands.

This applies to the other tests as well.

Reply:
see the commit message in https://github.com/web-platform-tests/wpt/pull/48851/commits/61810f37a9032123222de63a0e74b1adb53b1c4b:

> Let's avoid any risk of there being magic with how we manage sessions and directly interact with the transport, like we normally do while testing behaviour of commands.

in short, we do in some cases catch `InvalidSessionIdException` and change session state based on that, thus there is some benefit to avoiding all magic when we're trying to ensure we have no session

Reply:
Not sure I follow but reading the code at https://github.com/web-platform-tests/wpt/blame/master/tools/webdriver/webdriver/client.py#L560-L567 I can clearly see that we always re-raise the error - so it's not just in some cases. As such which other cases do we miss in your opinion?

Reply:
@gsnedders what do you think?